ir/expression.cpp

```diff
@@ -90,17 +90,32 @@ void IR::Constant::handleOverflow(bool noWarning) {
     big_int mask = Util::mask(width);
 
     if (tb->isSigned) {
+        auto masked_value = value;
+        bool is_overflow = false;
         big_int max = (one << (width - 1)) - 1;
         big_int min = -(one << (width - 1));
-        if (value < min || value > max) {
+
+        if (value >= 0) {
+            if (value != (value & mask)) {
+                is_overflow = true;
+                masked_value = value & mask;
+            }
+            if (masked_value > max) masked_value -= (one << width);
+        } else {
+            if (value < min) {
+                is_overflow = true;
+                masked_value = value & mask;
+            }
+        }
+        if (is_overflow) {
```

Comment:
This big block of code seems to be equivalent to just changing one line

    big_int max = (one << width) -1;

Reply:
Wouldn't this need to be adjusted based on signedness (`value >= 0`, `value <0`)?

@asawulaINTC Can you comment?